lib/rspec/core/reporter.rb

```diff
@@ -152,6 +152,18 @@ def deprecation(hash)
     end
 
     # @private
+    # Provides a way to notify of an exception that is not tied to any
+    # particular exception (such as an exception encountered in a :suite hook).
+    # Exceptions will be formatted the same way they normally are.
+    def notify_non_example_exception(exception, context_description)
+      @configuration.world.non_example_failure = true
+
+      example = Example.new(AnonymousExampleGroup, context_description, {})
+      presenter = Formatters::ExceptionPresenter.new(exception, example, :indentation => 0)
+      message presenter.fully_formatted(nil)
```

Comment:
`ExceptionPresenter` is tagged as `# @private` so it's not part of our public API.  I pass `nil` here because usually we pass an integer (e.g. `7` if it's the 7th example that has failed) but there isn't a good way to get a number here so the error just doesn't have the `7)` prefix.  a1936ff0332c82c1dbc0f331493750d7f5b45b77 was written to specifically allow this.